man/systemd.exec.xml

```diff
@@ -482,175 +482,181 @@
-        terminal.</para>
-
-        <para><option>tty-fail</option> is similar to
-        <option>tty</option> but if the terminal already has a
-        controlling process start-up of the executed process
-        fails.</para>
-
-        <para>The <option>socket</option> option is only valid in
-        socket-activated services, and only when the socket
-        configuration file (see
-        <citerefentry><refentrytitle>systemd.socket</refentrytitle><manvolnum>5</manvolnum></citerefentry>
-        for details) specifies a single socket only. If this option is
-        set, standard input will be connected to the socket the
-        service was activated from, which is primarily useful for
-        compatibility with daemons designed for use with the
-        traditional
-        <citerefentry project='freebsd'><refentrytitle>inetd</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+        <listitem><para>Controls where file descriptor 0 (STDIN) of the executed processes is connected to. Takes one
+        of <option>null</option>, <option>tty</option>, <option>tty-force</option>, <option>tty-fail</option>,
+        <option>data</option>, <option>file:<replaceable>path</replaceable></option>, <option>socket</option> or
+        <option>fd:<replaceable>name</replaceable></option>.</para>
+
+        <para>If <option>null</option> is selected, standard input will be connected to <filename>/dev/null</filename>,
+        i.e. all read attempts by the process will result in immediate EOF.</para>
+
+        <para>If <option>tty</option> is selected, standard input is connected to a TTY (as configured by
+        <varname>TTYPath=</varname>, see below) and the executed process becomes the controlling process of the
+        terminal. If the terminal is already being controlled by another process, the executed process waits until the
+        current controlling process releases the terminal.</para>
+
+        <para><option>tty-force</option> is similar to <option>tty</option>, but the executed process is forcefully and
+        immediately made the controlling process of the terminal, potentially removing previous controlling processes
+        from the terminal.</para>
+
+        <para><option>tty-fail</option> is similar to <option>tty</option>, but if the terminal already has a
+        controlling process start-up of the executed process fails.</para>
+
+        <para>The <option>data</option> option may be used to configure arbitrary textual or binary data to pass via
+        standard input to the executed process. The data to pass is configured via
+        <varname>StandardInputText=</varname>/<varname>StandardInputData=</varname> (see below). Note that the actual
+        file descriptor type passed (memory file, regular file, UNIX pipe, …) might depend on the kernel and available
+        privileges. In any case, the file descriptor is passed read-only, and when read returns the specified data
```

Comment:
"is <strike>passed</strike> read-only" (It's the fd itself that is readonly, there's nothing special in the way it is passed).